src/test/java/org/opensearch/knn/index/KNNSyntheticSourceIT.java

```diff
@@ -0,0 +1,159 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ *
+ * Modifications Copyright OpenSearch Contributors. See
+ * GitHub history for details.
+ */
+
+package org.opensearch.knn.index;
+
+import org.apache.hc.core5.http.ParseException;
+import org.apache.hc.core5.http.io.entity.EntityUtils;
+import org.opensearch.client.Response;
+import org.opensearch.common.settings.Settings;
+import org.opensearch.common.xcontent.XContentFactory;
+import org.opensearch.core.xcontent.XContentBuilder;
+import org.opensearch.knn.KNNRestTestCase;
+import org.opensearch.knn.common.KNNConstants;
+import org.opensearch.knn.index.query.KNNQueryBuilder;
+import org.opensearch.knn.index.util.KNNEngine;
+
+import java.io.IOException;
+import java.util.Map;
+
+public class KNNSyntheticSourceIT extends KNNRestTestCase {
+
+    public void testSyntheticSourceInSearch() throws IOException, ParseException {
```

Comment:
can we rename this function with this syntax: test_FuntionToTest_whenCondition_ThenResponse.

Reply:
FIXED at https://github.com/opensearch-project/k-NN/pull/1571/commits/2d16782f3590d015256fdc009595718013a35bb7